websites/D/Daum/presence.ts

```diff
@@ -0,0 +1,447 @@
+class DaumPresence extends Presence {
+  constructor(presenceOptions: PresenceOptions){
+    super(presenceOptions);
+  }
+
+  get serviceName(){
+    const url = document.location.hostname;
+
+    switch (true) {
+      case !!url.match(/movie([.]v)?[.]daum[.]([a-z0-9]+)/):
+        return "DAUM_MOVIE";
+      case !!url.match(/news([.]v)?[.]daum[.]([a-z0-9]+)/):
+        return "DAUM_NEWS";
+      case !!url.match(/sports([.]v)?[.]daum[.]([a-z0-9]+)/):
+        return "DAUM_SPORTS";
+      case !!url.match(/entertain([.]v)?[.]daum[.]([a-z0-9]+)/):
+        return "DAUM_ENTERTAIN";
+      case !!url.match(/blog([.]v)?[.]daum[.]([a-z0-9]+)/):
+        return "DAUM_BLOG";
+      case !!url.match(/auto([.]v)?[.]daum[.]([a-z0-9]+)/):
+        return "DAUM_AUTO";
+      case !!url.match(/mail([.]v)?[.]daum[.]([a-z0-9]+)/):
+        return "DAUM_MAIL";
+      case !!url.match(/(top[.])?cafe([.]v)?[.]daum[.]([a-z0-9]+)/):
+        return "DAUM_CAFE";
+      case !!url.match(/([a-z]+)([.]v)?[.]daum[.]([a-z0-9]+)/):
+        return "DAUM";
+      default:
+        break;
+    }
+  }
+
+  get isReadingArticle(){
+    return !!document.location.hostname.match(/([a-z]+)([.]v)[.]daum[.]([a-z0-9]+)/);
+  }
+}
+
+const presence = new DaumPresence({
+    clientId: "829266476875120650"
+}),
+  data: {
+    settings?: {
+      id: string,
+      delete?: boolean,
+      data: string[];
+    }[],
+    presence: {
+      [key: string]: {
+        service:
+          | "DAUM"
+          | "DAUM_MOVIE"
+          | "DAUM_AUTO"
+          | "DAUM_NEWS"
+          | "DAUM_SPORTS"
+          | "DAUM_BLOG"
+          | "DAUM_MAIL"
+          | "DAUM_ENTERTAIN"
+          | "DAUM_CAFE"
+          | "ANY";
+          href?: boolean,
+          hash?: boolean,
+          artitle?: boolean
+        setPresenceData?: () => void;
+      }
+    }
+  } = {
+    presence: null
+  },
+  startTime = Math.floor(Date.now() / 1000);
+
+let video : {
+  duration: number,
+  currentTime: number,
+  paused: boolean,
+  title?: string,
+},
+  cafe: {
+    name: string,
+    title: string,
+    article: string
+  };
+
+presence.on("iFrameData", (data: {
+  video?: typeof video,
+  cafe?: typeof cafe
+}) => { 
+  if (data.video) video = data.video;
+  if (data.cafe) cafe = data.cafe;
+});
+
+presence.on("UpdateData", async () => {
+    const presenceData: PresenceData = {
+      largeImageKey: "daum",
+      details: "Browsing...",
```

Comment:
is it really necessary when you did support for every subpage?

Reply:
idk but it just does looks cooler *xd*

Reply:
but ~~it doesn't~~

Reply:
meanie